core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -1541,6 +1541,36 @@ class KRaftClusterTest {
       cluster.close()
     }
   }
+
+  @Test
+  def testReduceNumNetworkThreads(): Unit = {
```

Comment:
Pardon me, is this test case related to deadlock?

Reply:
I wanted to ensure that no deadlocks are created in the future, by adding this test coverage. I don't think this PR is required to avoid deadlock currently, however.